api/shape/masking.js

```diff
@@ -0,0 +1,13 @@
+import Group from './Group.js';
+import Shape from './Shape.js';
+import { hasTypeMasked } from '@jsxcad/geometry';
+
+export const masking = (masked) => (shape) =>
```

Comment:
I don't think masking is documented, but I'm not sure how it works.

Reply:
Yeah, I've been a bit slack on that.

a.masking(b) === b.mask(a)